internal/hcp/internal/controllers/hcclink/controller.go

```diff
@@ -0,0 +1,80 @@
+package hcclink
+
+import (
+	"context"
+
+	"github.com/hashicorp/consul/internal/resource"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/consul/internal/controller"
+	pbhcp "github.com/hashicorp/consul/proto-public/pbhcp/v1"
+)
+
+func HCCLinkController(resourceApisEnabled bool, overrideResourceApisEnabledCheck bool) controller.Controller {
+	return controller.ForType(pbhcp.HCCLinkType).
+		WithReconciler(&hccLinkReconciler{
+			resourceApisEnabled:              resourceApisEnabled,
+			overrideResourceApisEnabledCheck: overrideResourceApisEnabledCheck,
+		})
```

Comment:
I _think_ we may also want to set the [placement](
https://github.com/hashicorp/consul/blob/main/docs/v2-architecture/controller-architecture/guide.md#placement
) of the controller to `PlacementEachServer` so that we (eventually) start an HCP manager on each server instance. 

We could also potentially update this later once we're actually starting the HCP connections when the HCCLink is created since I haven't confirmed if this is the behavior we need.

Reply:
Ok, I see. That makes sense.

> We could also potentially update this later

I agree — let's update this once we actually start doing work in the controller that requires that behavior.